src/app/portlets/dot-edit-page/content/components/dot-edit-page-workflows-actions/dot-edit-page-workflows-actions.component.ts

```diff
@@ -13,20 +14,47 @@ export class DotEditPageWorkflowsActionsComponent implements OnInit {
     @Input() inode: string;
     @Input() label: string;
 
-    workflowsActions: Observable<MenuItem[]>;
+    workflowsMenuActions: Observable<MenuItem[]>;
 
-    constructor(private workflowsService: DotWorkflowService) {}
+    constructor(private workflowsService: DotWorkflowService, private httpErrorManagerService: DotHttpErrorManagerService) {}
 
     ngOnInit() {
-        this.workflowsActions = this.workflowsService
-            .getContentWorkflowActions(this.inode)
-            .map((workflows: DotWorkflowAction[]) => this.getWorkflowOptions(workflows));
+        this.workflowsMenuActions = this.workflowsService.getContentWorkflowActions(this.inode).map((workflows: DotWorkflowAction[]) => {
+            return this.getWorkflowOptions(workflows);
+        });
     }
 
     private getWorkflowOptions(workflows: DotWorkflowAction[]): MenuItem[] {
         return workflows.map((workflow: DotWorkflowAction) => {
             return {
-                label: workflow.name
+                label: workflow.name,
+                command: () => {
+                    const currentMenuActions = this.workflowsMenuActions;
+                    this.workflowsMenuActions = this.workflowsService
+                        .fireWorkflowAction(this.inode, workflow.id)
+                        .pluck('inode')
+                        .catch((error) => Observable.of(null))
+                        .mergeMap((inode: string) => {
+                            this.inode = inode || this.inode;
```

Comment:
if the fireWorkflowAction method throw a error then we just reload the before inode's workflow actions?

Reply:
Yes

Reply:
I think we should add a TODO here, this is not a good solution in the case the fire request catch